### PR TITLE
User search bug fixed

### LINKED ISSUE #944: Search with full name is not working
URL: https://billing.faveohelpdesk.com/clients In name field Search for Ron Brown, search result 0
But client exist in the system https://billing.faveohelpdesk.com/clients/4758

### PR DESCRIPTION
**Pull request by ashu555**
Fixes #944

### CODE REVIEW

File: app/Http/Controllers/User/AdvanceSearchController.php [hunk 1]
```diff
@@ -91,8 +91,10 @@ public function getregFromTill($join, $reg_from, $reg_till)
     public function getNamUserCom($join, $name, $username, $company)
     {
         if ($name) {
-            $join = $join->where('first_name', 'LIKE', '%'.$name.'%')
-                    ->orWhere('last_name', 'LIKE', '%'.$name.'%');
+            $join->where(function ($q) use ($name) {
+                $name = trim($name);
+                $q->whereRaw("concat(first_name, ' ', last_name) like '%$name%' ");;
```
Comment (avinash403) [inline app/Http/Controllers/User/AdvanceSearchController.php:96]: its open for SQL injection
make it like 
```
                 $q->whereRaw("concat(first_name, ' ', last_name) like ?", ["%$name%"]);
```

Comment (avinash403) [reply app/Http/Controllers/User/AdvanceSearchController.php:96]: please ask me if you didn't understand why we need this

**Comment by bindu95** [comment]
@ashu555 can you please comment how this search will work now?

**Comment by bindu95** [comment]
@ashu555 Not able to replicate
Tried in master, development and bug fix branch
https://docs.google.com/document/d/1Kj0xd1EKnzf4yTrQheiBUsvltCJ31mJbUrp1yBGGKCQ/edit?usp=sharing
Please let me know the issue

**Pull request closed without merging**